### PR TITLE
Use wp_remote functions instead of file_get_contents

### PR DESCRIPTION
**Pull request by nigelbabu**
This is a more standard way of accessing remote resources than
`file_get_contents`. Some hosting providers turn off the
`allow_url_fopen` option, limiting using this plugin properly.

### CODE REVIEW

**Comment by retlehs** [comment]
don't need the custom function for remote get - wordpress core has a function for this. wp_remote_get i believe

**Comment by nigelbabu** [comment]
@retlehs Thanks for that! I've just made the changes. Much cleaner pull request :)

File: lib/discourse.php [hunk 1]
```diff
@@ -466,17 +464,13 @@ function sync_to_discourse_work( $postid, $title, $raw ) {
...
       // use key 'http' even if you send the request to https://...
-      $soptions = array(
-        'http' => array(
-          'ignore_errors' => true,
-          'method'  => 'POST',
-          'content' => http_build_query( $data ),
-          'header' => "Content-Type: application/x-www-form-urlencoded\r\n"
-        )
+      $post_options = array(
+        'timeout'     => 30,
```
Comment (retlehs) [inline lib/discourse.php]: remove this extra spacing, then squash these commits into one and this will be ready to merge

**Comment by nigelbabu** [comment]
@retlehs All done :)

**Comment by retlehs** [comment]
@techAPJ if you have a chance to test this and it works, it's good to go :+1:

**Comment by nigelbabu** [comment]
@retlehs I've got some code to catch HTTP errors and log it. Kind of useful for the occasional large post being sent over. Would you like me to push that with this?

**Comment by techAPJ** [comment]
@nigelbabu: let me quickly test and merge this PR, so that you can create a new PR for logging.

**Comment by techAPJ** [comment]
Thanks!  :+1:

**Pull request merged**